### PR TITLE
Restucture documents and setup mdbook Github Action

### PR DESCRIPTION
**Pull request by andylokandy**
Signed-off-by: Andy Lok <andylokandy@hotmail.com>

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-- [Transactions overview](doc/README.md)
-- [Transactions in TiKV](doc/tikv/README.md)
+- [Transactions in TiKV](doc/tikv/transaction-in-tikv.md)
+- [Transactions handling newbiew perspective](doc/tikv/transaction-handling-newbie-perspective/transaction-handling-newbie-perspective.md)
 
 # Design docs
 
 - [Proposal for refactoring how transactional commands are scheduled and implemented](design/transaction-layer-refactoring.md)
-- [Async commit](design/async-commit/README.md)
+- [Async commit](design/async-commit/async-commit.md)
```
Comment (sticnarf) [inline SUMMARY.md:13]: Calling it `README.md` makes it render for `design/async-commit` directory.

Comment (andylokandy) [reply SUMMARY.md:13]: done

Comment (andylokandy) [reply SUMMARY.md]: done

**Comment by sticnarf** [comment]
@andylokandy You mistype README as READMD   XD

File: design/async-commit/README.md [hunk 2]
```diff
@@ -154,15 +154,15 @@ TODO
...
+There are restrictions on the size of the transaction. For example, if the key involved in the transaction is less than 64, async commit is used, or a hierarchical structure is adopted. The primary lock records a few secondary locks, and these secondary locks record other secondary locks respectively. It is easy to implement, just recursion, and the cost of failure recovery needs to be considered.
 
 Crdb mentioned two ways to reduce the impact of recovery, and TiDB has also implemented: one is to perform commit cleanup as soon as possible when committing; the second is transaction heartbeat to prevent cleanup of alive transactions.
 
 ## Related Work
 
 ### Cockroach DB
 
-In CRDB, parallel commit extends pipelined pessimistic locking.
+In CRDB, async commit extends pipelined pessimistic locking.
```
Comment (sticnarf) [inline design/async-commit/README.md:165]: CockroachDB uses the wording "parallel commit".

Comment (andylokandy) [reply design/async-commit/README.md:165]: 👌

**Review by sticnarf — APPROVED** [review]

**Pull request merged**